src/IROperator.cpp

```diff
@@ -537,6 +538,7 @@ Expr lossless_cast(Type t, Expr e) {
                     Type narrower = reduce->value.type().with_bits(t.bits() / 2);
                     Expr val = lossless_cast(narrower, reduce->value);
                     if (val.defined()) {
+                        val = cast(narrower.with_bits(t.bits()), val);
```

Comment:
Is this a bug-fix on master? The original looks wrong to me now.

Reply:
Yes. I encountered this error while working on the PR.